internal/stubserver/stubserver.go

```diff
@@ -56,9 +56,10 @@ type StubServer struct {
 
 	// Parameters for Listen and Dial. Defaults will be used if these are empty
 	// before Start.
-	Network string
-	Address string
-	Target  string
+	Network  string
+	Address  string
+	Target   string
+	Listener net.Listener
```

Comment:
This is not a parameter for `Listen` or `Dial`.  Can you document this please and indicate `Network` & `Address` aren't used if this is set, and that `Target` is the responsibility of the user to set properly when `Listener` is used?

Reply:
My code search reveals that nobody is setting `Network`, `Address` or `Target`. 

`Address` is used as a read-only field by numerous tests to retrieve the address of the stub server. Should we just get rid of `Network` and `Target` and make `Address` as a read-only field, or make it a getter method? (Can do in a follow up PR)

Reply:
SGTM!

Reply:
Will do the cleanup in a follow up PR. Thanks.